winpr/libwinpr/crypto/hash.c

```diff
@@ -241,7 +241,7 @@ BOOL winpr_HMAC_Init(WINPR_HMAC_CTX* ctx, WINPR_MD_TYPE md, const void* key, siz
 		return FALSE;
 
 	const char* param_name = OSSL_MAC_PARAM_DIGEST;
-	const OSSL_PARAM param[] = { OSSL_PARAM_construct_utf8_string(param_name, hash, 0),
+	const OSSL_PARAM param[] = { OSSL_PARAM_construct_utf8_string(param_name, (char*)hash, 0),
```

Comment:
will now complain with a `warning: cast from 'const char *' to 'char *' drops const qualifier [-Wcast-qual]`, you will not get this silent unless `OpenSSL` provides a `const` correct version of this function :/

Reply:
It seems not only this function suffers from it: https://github.com/openssl/openssl/issues/22535

I've noticed that the Clang `-Wincompatible-pointer-types-discards-qualifiers` warning is emitted without any `-Wall` or `-Wextra`, while it's now gone with the cast in place. The `-Wcast-qual` also doesn't seem to be enabled on GCC with `-Wall -Wextra`. Not saying it isn't a valid warning, however I'd consider the situation with the cast better than without the cast.

Another option would be to use `strdup()` here, but that would make things really, really complex for solving a warning: not only do you have to remember to call free() on the early return before `OSSL_PARAM_construct_utf8_string`, but you also have to know what happens with the `strdup()`'ed string after you pass it to OpenSSL. Can it be safely freed after calling `EVP_MAC_init()`? I'm not sure. If you have some guidance there, maybe we could make that work, seeing as others (from the linked OpenSSL issue) have also solved this with `strdup()`.

Open to suggestions :)